eng/versioning/set_versions.py

```diff
@@ -93,23 +109,133 @@ def prep_version_file_for_source_testing(build_type):
             f.write(line)
     
     return file_changed
-    
+
+# given a build type and artifact id
+def increase_version_for_artifact(build_type, artifact_id):
+
+    if not build_type:
+        raise ValueError('build_type cannot be empty.')
+
+    if not artifact_id:
+        raise ValueError('artifact_id cannot be empty.')
+
+    version_file = os.path.normpath('eng/versioning/version_' + build_type.name + '.txt')
+    print('version_file=' + version_file)
+
+    artifact_found = False
+    newlines = []
+    with open(version_file, encoding='utf-8') as f:
+        for raw_line in f:
+            stripped_line = raw_line.strip()
+            if not stripped_line or stripped_line.startswith('#'):
+                newlines.append(raw_line)
+            else:
+                module = CodeModule(stripped_line)
+                # Tick up the version here. If the version is already a pre-release
+                # version then just increment the revision. Otherwise increment the
+                # hotfix version and add the prerelease tag "-beta.1" to the end
+                if module.artifact_id == artifact_id:
+                    artifact_found = True
+                    vmatch = version_regex_named.match(module.current)
+                    if (vmatch.group('prerelease') is not None):
+                        prever = prerelease_regex_named.match(vmatch.group('prerelease'))
+                        rev = int(prever.group('revision'))
+                        rev += 1
+                        new_version = '{}.{}.{}-beta.{}'.format(vmatch.group('major'), vmatch.group('minor'), vmatch.group('patch'), str(rev))
+                    else:
+                        hotfix = int(vmatch.group('patch'))
```

Comment:
This should increment the minor version by default. https://github.com/Azure/azure-sdk/blob/master/docs/policies/releases.md#java

Reply:
The next commit will have this behavior as well as the link to the doc.